### DISCUSSION TITLE
Heltec wireless stick lite v3 auto select problem

### DISCUSSION THREAD

**Discussion by vladosam** [jgromes/RadioLib#1118] [General]
Hi, I have problem with auto selecting mcu with my heltec board. I'm using starter lorawan example and on compile i'm getting "Unknown board - no automagic pinmap available". On 6.4.2 radiolib version I have this for my board "SX1262 radio = new Module(8, 14, 12, 13);" and if I add that to config.h it compiles. But in config.h there is section for Heltec wireless stick lite v3 board and it has other pinmap
```
#elif defined (ARDUINO_HELTEC_WIRELESS_STICK_LITE_V3)
  #pragma message ("Using Heltec Wireless Stick Lite v3")
  SX1262 radio = new Module(34, 14, 12, 13);
```
How can I use auto selecting mcu for this board, do I need to set something in Arduino IDE or in my code? Thanks.

**Comment by StevenCellist** [comment]
The lazy option for now is to add this line in `config.h` for now:
```cpp
#define ARDUINO_HELTEC_WIRELESS_STICK_LITE_V3
```

There are some problems with the auto-select which are being resolved through #1106. In the next update of RadioLib (although that will probably take a while) this is likely going to be integrated instead of whatever is currently happening in `config.h`. So for the time being, I'd recommend not bothering too much and adding the define yourself.

**Reply by StevenCellist** [reply]
To be clear: if you select the Wireless Stick Lite V3 as per the Heltec-esp32 package, it should work I think. But if it doesn't, meh, not worth the time probably, it's so much easier to add the definition yourself.

**Reply by vladosam** [reply]
Thanks, adding #define ARDUINO_HELTEC_WIRELESS_STICK_LITE_V3 sets board, but after upload I'm getting
```
Setup ... 
Initialise the radio
Initialise radio failed - ERR_CHIP_NOT_FOUND (-2)
```
I guess that is because pinmap for HWSLV3 on config.h is 
```
  SX1262 radio = new Module(34, 14, 12, 13);
```
and on my old code for radiolib 6.4.2 pinmap and spi setting is
```
#include <SPI.h>
SPIClass spi(SPI);
SPISettings spiSettings(2000000, MSBFIRST, SPI_MODE0);
// SX1262 has the following pin order:
// Module(NSS/CS, DIO1, RESET, BUSY)
SX1262 radio = new Module(8, 14, 12, 13, spi, spiSettings);
```
Are there other Heltec Wireless Stick Lite V3 board different pinmaps out there?

**Comment by Velocet** [comment]
May i suggest to always use [Arduino-ESP32](https://github.com/espressif/arduino-esp32) instead of anything from HelTec. There is a reason @ropg made an unofficial [HelTec LoRa32 V3](https://github.com/ropg/heltec_esp32_lora_v3) library.

Is it possible to add a note in the code base as general advice?

**Reply by StevenCellist** [reply]
It may be possible, but with the upcoming `RadioBoards` it will work just fine. So there should be no need for it and is subsequently outside the scope of RadioLib, I'd say. Wish I could pin it as a message on the Heltec forum...

**Reply by vladosam** [reply]
> May i suggest to always use [Arduino-ESP32](https://github.com/espressif/arduino-esp32) instead of anything from HelTec. There is a reason @ropg made an unofficial [HelTec LoRa32 V3](https://github.com/ropg/heltec_esp32_lora_v3) library.
> 
> Is it possible to add a note in the code base as general advice?

I want to use Arduino-ESP32 but radio initialisation failed until I changed board to Heltec Wireless Stick Lite V3. Also changed pinmap to 
```
  SX1262 radio = new Module(8, 14, 12, 13);
```
On 6.4.2 I could use ESP32S3-DEV board in Arduino-IDE.
